serde_arrow/benches/groups/json_to_arrow.rs

```diff
@@ -0,0 +1,101 @@
+use crate::groups::complex_common::Item;
+
+use {
+    serde_arrow::schema::{SchemaLike as _, SerdeArrowSchema},
+    std::sync::Arc,
+};
+
+// arrow-version:replace: use arrow_json_{version}::ReaderBuilder;
+use arrow_json_50::ReaderBuilder;
+
+// arrow-version:replace: use arrow_schema_{version}::Schema as ArrowSchema;
+use arrow_schema_50::Schema as ArrowSchema;
+
+// arrow-version:replace: use arrow_array_{version}::RecordBatch;
+use arrow_array_50::RecordBatch;
+
+fn benchmark_json_to_arrow(c: &mut criterion::Criterion) {
+    let rng = &mut rand::thread_rng();
+    let items = (0..10_000)
+        .map(|_| Item::random(rng))
+        .collect::<Vec<Item>>();
+    let jsons_to_deserialize = items
+        .iter()
+        .map(|item| serde_json::to_string(item).expect("Failed to serialize JSON"))
+        .collect::<Vec<_>>();
+    let jsons_to_deserialize_concatenated = jsons_to_deserialize.join("\n");
+    let jsons_to_deserialize: Vec<&str> = {
+        let mut prev = 0;
+        jsons_to_deserialize
+            .iter()
+            .map(|s| {
+                let ret = &jsons_to_deserialize_concatenated[prev..(prev + s.len())];
+                prev += s.len() + 1;
+                ret
+            })
+            .collect::<Vec<_>>()
+    };
+    let schema = SerdeArrowSchema::from_samples(&items, Default::default()).unwrap();
+    let arrow_fields = schema.to_arrow_fields().unwrap();
+    let mut group = c.benchmark_group(format!("json_to_arrow({})", items.len()));
+
+    // arrow-json direct
+    group.bench_function("arrow-json", |b| {
+        b.iter(|| {
+            let schema = Arc::new(ArrowSchema::new(arrow_fields.to_owned()));
+            let mut decoder = ReaderBuilder::new(schema.clone()).build_decoder().unwrap();
+            decoder
+                .decode(criterion::black_box(
```

Comment:
black_box should probably wrapped around the record batch, as otherwise the compiler is free to optimize away record batch construction. Which is a known issue for serde-arrow